sentry-android-core/src/main/java/io/sentry/android/core/ScreenshotEventProcessor.java

```diff
@@ -0,0 +1,140 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import android.app.Application;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.os.Bundle;
+import android.view.View;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import io.sentry.Attachment;
+import io.sentry.EventProcessor;
+import io.sentry.SentryEvent;
+import io.sentry.SentryLevel;
+import io.sentry.util.Objects;
+import java.io.ByteArrayOutputStream;
+import java.io.Closeable;
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+
+public final class ScreenshotEventProcessor
+    implements EventProcessor, Application.ActivityLifecycleCallbacks, Closeable {
+
+  private final @NotNull Application application;
+  private final @NotNull SentryAndroidOptions options;
+  private @Nullable WeakReference<Activity> currentActivity;
+
+  public ScreenshotEventProcessor(
+      final @NotNull Application application, final @NotNull SentryAndroidOptions options) {
+    this.application = Objects.requireNonNull(application, "Application is required");
+    this.options = Objects.requireNonNull(options, "SentryAndroidOptions is required");
+
+    if (this.options.isAttachScreenshot()) {
+      application.registerActivityLifecycleCallbacks(this);
+    }
+  }
+
+  @Override
+  public @NotNull SentryEvent process(
+      final @NotNull SentryEvent event, @Nullable Map<String, Object> hint) {
+    if (options.isAttachScreenshot() && event.isErrored() && currentActivity != null) {
+      final Activity activity = currentActivity.get();
+      if (activity != null
+          && activity.getWindow() != null
+          && activity.getWindow().getDecorView() != null
+          && activity.getWindow().getDecorView().getRootView() != null) {
+        final View view = activity.getWindow().getDecorView().getRootView();
+
+        if (view.getWidth() > 0 && view.getHeight() > 0) {
+          try {
+            // ARGB_8888 -> This configuration is very flexible and offers the best quality
+            final Bitmap bitmap =
+                Bitmap.createBitmap(view.getWidth(), view.getHeight(), Bitmap.Config.ARGB_8888);
+
+            final Canvas canvas = new Canvas(bitmap);
```

Comment:
Canvas+Bitmap allow taking the screenshot even not in the main thread, so I've removed the main thread checker.